src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -377,10 +377,16 @@ private Pair<ConnectionString, DriverPropertyInfo[]> getConnectionSettings(
         char[] password = result.password;
 
         List<DriverPropertyInfo> mandatoryConnectionProperties = new ArrayList<>();
+
+        // A database to connect to is required. If they have not specified one, look in the connection string for a
```

Comment:
The only part which worries me a bit is this: `String authDatabase = originalConnectionString.getDatabase();`
I believe this means that we never use the actual authSource setting a user provides.
This will depend on what the code at the bottom of this function does
```
                new ConnectionString(
                        buildNewURI(
                                originalConnectionString.getHosts(),
                                user,
                                password,
                                authDatabase,
                                result.normalizedOptions));
```